gap/congruences/congsemigraph.gi

```diff
@@ -0,0 +1,271 @@
+############################################################################
+##
+##  congsemigraph.gi
+##  Copyright (C) 2021                       Marina Anagnostopoulou-Merkouri
+##                                                            James Mitchell
+##
+##  Licensing information can be found in the README file of this package.
+##
+############################################################################
+
+BindGlobal("SEMIGROUPS_IsHereditarySubset",
+function(S, H)
+  local out, h, v, D;
+  D := GraphOfGraphInverseSemigroup(S);
+  if IsMultiDigraph(D) then
+    ErrorNoReturn(D, " must not have multiple edges");
+  fi;
+  out := OutNeighbours(D);
+  for h in H do
+    for v in out[h] do
+      if not (v in H) then
```

Comment:
It might make sense to replace `H` here by (a variable assigned to) `BlistList(DigraphVertices(D), H);` so that this is constant time, and not `Length(H)` time. It might also make sense to use a `HashSet` if the number of vertices in `D` is large and `H` is small, but I'm not sure if this is worth it. You might also want to add the special case of `H = DigraphVertices(D)`, in which case you can just return `true`, again I'm not sure if this case will ever arise, if not, then you can ignore this part of the comment.

Reply:
@james-d-mitchell if I use a BlistList, wouldn't I have to then iterate over all the vertices `v` and check whether `BlistList(DigraphVertices(D), H)[v]` is true? Is there another way to check existence in `H` using that list without iterating over the vertices of the graph that I'm not aware of?